src/App.tsx

```diff
@@ -7,33 +9,84 @@ function getRandomName(): string {
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean,
+  clockName: string,
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timerId = 0;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+    document.addEventListener('click', this.handleLeftMouseButton);
+    document.addEventListener('contextmenu', this.handleRightMouseButton);
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentWillUnmount() {
+    window.clearInterval(this.timerId);
+    document.removeEventListener('click', this.handleLeftMouseButton);
+    document.removeEventListener('contextmenu', this.handleRightMouseButton);
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  handleLeftMouseButton = () => {
+    this.setState({ hasClock: true });
+  };
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  handleRightMouseButton = (event: MouseEvent) => {
+    event.preventDefault();
+    this.setState({ hasClock: false });
+  };
 
-        {' time is '}
+  render() {
+    const { hasClock, clockName } = this.state;
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+
+        {hasClock && (
+          <Clock name={clockName} />
+        )}
       </div>
-    </div>
-  );
-};
+    );
+  }
+}
+
+// export const App: React.FC = () => {
+//   const today = new Date();
+//   let clockName = 'Clock-0';
+
+//   // This code starts a timer
+//   const timerId = window.setInterval(() => {
+//     clockName = getRandomName();
+//   }, 3300);
+
+//   // this code stops the timer
+//   window.clearInterval(timerId);
+
+//   return (
+//     <div className="App">
+//       <h1>React clock</h1>
+
+//       <div className="Clock">
+//         <strong className="Clock__name">
+//           {clockName}
+//         </strong>
+
+//         {' time is '}
+
+//         <span className="Clock__time">
+//           {today.toUTCString().slice(-12, -4)}
+//         </span>
+//       </div>
+//     </div>
+//   );
+// };
```

Comment:
Mb delete this comments

Reply:
my bad, i used these comments such as sketch)